libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -833,20 +828,12 @@ string IRGenerator::deployCode(ContractDefinition const& _contract)
 		{
 			solUnimplementedAssert(immutable->type()->isValueType(), "");
 			solUnimplementedAssert(immutable->type()->sizeOnStack() == 1, "");
-			string yulVar = m_context.newYulVariable();
-			loadImmutables.emplace_back(map<string, string>{
-				{"var"s, yulVar},
-				{"memoryOffset"s, to_string(m_context.immutableMemoryOffset(*immutable))}
-			});
-			storeImmutables.emplace_back(map<string, string>{
-				{"var"s, yulVar},
-				{"immutableName"s, to_string(immutable->id())}
+			immutables.emplace_back(map<string, string>{
+				{"immutableName"s, to_string(immutable->id())},
+				{"value"s, "mload(" + to_string(m_context.immutableMemoryOffset(*immutable)) + ")"}
```

Comment:
Oh I see, the whole point of this immutable refactoring is so that no intermediate variable needs to be allocated. It is not connected to the memory model change.

Reply:
It is: The memory change allows us to do that in the first place, because we do not overwrite the current values of the immutables by the codecopy.

Reply:
Ah right, due to the potential overlap.